jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Server.java

```diff
@@ -700,9 +729,12 @@ protected void doStop() throws Exception
         if (getStopAtShutdown())
             ShutdownThread.deregister(this);
 
-        //Unregister the Server with the handler thread for receiving
-        //remote stop commands as we are stopped already
-        ShutdownMonitor.deregister(this);
+        Shutdown shutdown = getBean(Shutdown.class);
```

Comment:
Just do both ShtdownMonitor.deregister and getBean(Shutdown.class).removeComponent(this)

Reply:
Why have 2 Shutdown controlling servers active at the same time?